lighthouse-core/lib/page-functions.js

```diff
@@ -6,8 +6,39 @@
 // @ts-nocheck
 'use strict';
 
+/** @typedef {HTMLElementTagNameMap & {[id: string]: HTMLElement}} HTMLElementByTagName */
+
 /* global window document Node ShadowRoot */
 
+/**
+ * Creates valid JavaScript code given functions, strings of valid code, and arguments.
+ * @template T, R
+ * @param {(...args: T[]) => R} mainFn The main function to call. It's return value will be the return value
+ * of `createEvalCode`, wrapped in a Promise.
+ * @param {{mode?: 'iife'|'function', args?: T[], deps?: Array<Function|string>}} _ Set mode to `iife` to
+ * create a self-executing function expression, set to `function` to create just a function
+ * declaration statement. Args should match the args of `mainFn`, and can be any serializable
+ * value. `deps` are functions that must be defined for `mainFn` to work.
+ */
+function createEvalCode(mainFn, {mode, args, deps} = {}) {
+  const argsSerialized = args ? args.map(arg => JSON.stringify(arg)).join(',') : '';
+  const depsSerialized = deps ? deps.join('\n') : '';
```

Comment:
nit: `join()` on an empty list is an empty string anyway. You can get rid of the conditionals.

Reply:
args and deps are optional params.

strangely, typecheck found no error when I removed the conditionals, even though both were marked as `Array | undefined` ...

Reply:
> strangely, typecheck found no error when I removed the conditionals, even though both were marked as `Array | undefined` ...

it's because this file is `// @ts-nocheck`. Maybe after this change and followup(s), we could drop that with a minimum of expect-errors